File: 29. Divide Two Integers/divide_two_integers.js
function division_whole (number, divisor){
	let out = 0;
	let running_number = number
	
	while ((running_number - divisor) >= 0){
		out++;
		running_number -= divisor;
	}
	
	return (out);
}

function modulo_whole (number, divisor){
	let running_number = number;
	
	while ((running_number - divisor) >= 0){
		running_number -= divisor;
	}
	
	return (running_number);
}

function multiplication (multiplicand, multiplier){
	let out = 0;
	
	for (let i = 0; i < multiplier; i++){
		out += multiplicand;
	}
	
	return out;
}

function division (number, divisor, no_of_figures = 10){
	let base = 10;
	let counter = 0;
	let fraction = 0;
	let integer_quotient = division_whole(number, divisor);
	let remainder = modulo_whole(number, divisor);
	let running_number = remainder;
	let out_of = 1;
	
	while ((running_number != 0) && (counter < no_of_figures)){
		//running_number *= base;
		running_number = multiplication(running_number, base);
		//out_of *= base
		out_of = multiplication(out_of, base);
		
		if (fraction == 0){
			fraction += division_whole(running_number, divisor);
		} else {
			//fraction *= base
			fraction = multiplication(fraction, base);
			fraction += division_whole(running_number, divisor);
		}
		
		running_number = modulo_whole(running_number, divisor)
		counter += 1;
	}
	
	return [integer_quotient, fraction, out_of];
}


number = 6402
divisor = 8

console.log(division(number, divisor));
//console.log(modulo(number, divisor));
//console.log(division_fractiontion(7, divisor));

File: 29. Divide Two Integers/divide_two_integers.rb
def division_whole (number, divisor)
	out = 0
	running_number = number
	
	while ((running_number - divisor) >= 0)
		out += 1
		running_number -= divisor
	end
	
	return out
end

def modulo_whole (number, divisor)
	running_number = number
	
	while ((running_number - divisor) >= 0)
		running_number -= divisor
	end
	
	return running_number
end

def multiplication (multiplicand, multiplier)
	out = 0
	
	for i in 0...multiplier
		out += multiplicand
	end
	
	return out
end

def division (number, divisor, no_of_figures = 10)
	base = 10
	counter = 0
	fraction = 0
	integer_quotient = division_whole(number, divisor)
	remainder = modulo_whole(number, divisor)
	running_number = remainder
	out_of = 1
	
	while ((running_number != 0) and (counter < no_of_figures))
		#running_number *= base
		running_number = multiplication(running_number, base)
		#out_of *= base
		out_of = multiplication(out_of, base)
		
		if (fraction == 0)
			fraction += division_whole(running_number, divisor)
		elsif
			#fraction *= base
			fraction = multiplication(fraction, base)
			fraction += division_whole(running_number, divisor)
		end
		
		running_number = modulo_whole(running_number, divisor)
		
		counter += 1
		
	end
	
	return [integer_quotient, fraction, out_of]
end

number = 6402
divisor = 8

puts (division(number, divisor))
#puts (modulo(number, divisor))
#puts (division_fraction(7,divisor))
